ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsInitializeIngestion.scala

```diff
@@ -26,234 +27,297 @@ private[backend] trait StorageBackendTestsInitializeIngestion
+      // 1: transaction with a create node
+      dtoCreate(offset(1), 1L, hashCid("#101"), signatory = signatory),
+      DbDto.IdFilterCreateStakeholder(1L, someTemplateId.toString, someParty),
+      DbDto.IdFilterCreateNonStakeholderInformee(1L, someParty),
+      dtoTransactionMeta(
+        offset(1),
+        event_sequential_id_first = 1L,
+        event_sequential_id_last = 1L,
+      ),
+      dtoCompletion(offset(41)),
+      // 2: transaction with exercise node and retroactive divulgence
+      dtoExercise(offset(2), 2L, false, hashCid("#101")),
+      DbDto.IdFilterNonConsumingInformee(2L, someParty),
+      dtoExercise(offset(2), 3L, true, hashCid("#102")),
+      DbDto.IdFilterConsumingStakeholder(3L, someTemplateId.toString, someParty),
+      DbDto.IdFilterConsumingNonStakeholderInformee(3L, someParty),
+      dtoDivulgence(Some(offset(2)), 4L, hashCid("#101")),
+      dtoTransactionMeta(
+        offset(2),
+        event_sequential_id_first = 2L,
+        event_sequential_id_last = 4L,
+      ),
+      dtoCompletion(offset(2)),
     )
 
-    val readers = Set(signatory)
+    it should "delete overspill entries - events, transaction meta, completions" in {
+      fixture(
+        dtos1 = dtos,
+        lastOffset1 = 2L,
+        lastEventSeqId1 = 4L,
+        dtos2 = Vector(
+          // 3: transaction with create node
+          dtoCreate(offset(3), 5L, hashCid("#201"), signatory = signatory),
+          DbDto.IdFilterCreateStakeholder(5L, someTemplateId.toString, someParty),
+          DbDto.IdFilterCreateNonStakeholderInformee(5L, someParty),
+          dtoTransactionMeta(
+            offset(3),
+            event_sequential_id_first = 5L,
+            event_sequential_id_last = 5L,
+          ),
+          dtoCompletion(offset(3)),
+          // 4: transaction with exercise node and retroactive divulgence
+          dtoExercise(offset(4), 6L, false, hashCid("#201")),
+          DbDto.IdFilterNonConsumingInformee(6L, someParty),
+          dtoExercise(offset(4), 7L, true, hashCid("#202")),
+          DbDto.IdFilterConsumingStakeholder(7L, someTemplateId.toString, someParty),
+          DbDto.IdFilterConsumingNonStakeholderInformee(7L, someParty),
+          dtoDivulgence(Some(offset(4)), 8L, hashCid("#201")),
+          dtoTransactionMeta(
+            offset(4),
+            event_sequential_id_first = 6L,
+            event_sequential_id_last = 8L,
+          ),
+          dtoCompletion(offset(4)),
+        ),
+        lastOffset2 = 10L,
+        lastEventSeqId2 = 6L,
+        checkContentsBefore = () => {
+          val contract101 =
+            executeSql(backend.contract.activeContractWithoutArgument(readers, hashCid("#101")))
+          val contract202 =
+            executeSql(backend.contract.activeContractWithoutArgument(readers, hashCid("#201")))
+          // TODO etq: Add assertion for the remaining filter tables & transaction_meta table
+          val idsCreateStakeholder = executeSql(
```

Comment:
fyi @mziolekda - I modified tests in this class to cover the copy-paste mistake you spotted during review, but we can't just yet assert on the contents of the new tables because we don't have queries for that yet.

(I also refactored these tests because their size and copy-paste nature was getting out of hand)